composer/cli/launcher.py

```diff
@@ -272,6 +273,13 @@ def _get_file(format: str):
             process.stdout = stdout_file
         processes[global_rank] = process
 
+        # Restore the env variables back to the original values
+        for var in ("RANK", "WORLD_SIZE", "LOCAL_RANK", "LOCAL_WORLD_SIZE", "NODE_RANK", "MASTER_ADDR", "MASTER_PORT"):
+            if var in original_env:
+                os.environ[var] = original_env[var]
+            else:
+                del os.environ[var]
```

Comment:
Should we be doing this? Or just restoring the distributed-related envvars? Regradless, there should be a `log.info` here

Reply:
(I re-wrote this with a context manager to make it clearer what is happening)

Reply:
LGTM